tools/azure-sdk-tools/devtools_testutils/storage/testcase.py

```diff
@@ -180,15 +181,8 @@ def assert_download_progress(self, size, max_chunk_size, max_get_size, progress)
                 assert i[1] == size
 
     def generate_oauth_token(self):
-        if self.is_live:
-            from azure.identity import ClientSecretCredential
-
-            return ClientSecretCredential(
-                self.get_settings_value("TENANT_ID"),
-                self.get_settings_value("CLIENT_ID"),
-                self.get_settings_value("CLIENT_SECRET"),
-            )
-        return self.generate_fake_token()
+        cred = self.get_credential(BlobServiceClient)
+        return cred
 
     def generate_fake_token(self):
```

Comment:
Can remove this in the future since `self.get_credential()` handles recorded credentials as well!